exporter/datadogexporter/internal/logsagent/agent.go

```diff
@@ -0,0 +1,150 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package logsagent // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/logsagent"
```

Comment:
We need to move this to DD Agent code base. We need to need similar agent implementation.

Reply:
Moved in this PR: https://github.com/DataDog/datadog-agent/pull/25032